### PR TITLE
Add Youtube, Vimeo, Video (both) validator

### PR DESCRIPTION
**Pull request by ricardogobbosouza**

### CODE REVIEW

**Comment by henriquemoody** [comment]
Cool! Thanks for contributing! :panda_face: 

I'm just confusing about the names, because a _video_ is not the same of a _video URL_, a _Youtube video_ is not the same of _Youtube_ and so on... Could you rename them?

File: library/Rules/Video.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace Respect\Validation\Rules;
+
+class Video extends AbstractRule
+{
+    public function validate($input)
+    {
+        return !empty($input) && (
+            preg_match('/^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#\&\?]*).*/i', $input) ||
+            preg_match('/^.*(vimeo\.com\/)((channels\/[A-z]+\/)|(groups\/[A-z]+\/videos\/)|(video\/))?([0-9]+)/i', $input)
```
Comment (henriquemoody) [inline library/Rules/Video.php]: Here and in other patterns you allow any protocol, it that what you indented to do?

Comment (ricardogobbosouza) [inline library/Rules/Video.php]: They are validators URL:

Youtube - Accepts URL Youtuve
Vimeo - Accepts URL Vimeo
Video - Supports both URL

I would like to move to UrlYoutube?

Sorry my bad engilish.

Comment (henriquemoody) [inline library/Rules/Video.php]: I would probably go for only one rule, called `videoUrl()` which accepts argument on its construtor, that can be `"youtube"` or `"vimeo"`, them we can implement more, like `"ted"`, `"viki"`... You can create a key => value array with the patterns to match for the defined video provider (youtube, vimeo...) or try all of them if no provider is defined.

Comment (ricardogobbosouza) [inline library/Rules/Video.php]: Ok. I'm changing ..
Thanks

**Comment by henriquemoody** [comment]
Merged manually by a0261245904b8dc48ec3c6a23648d0d41e6575b6

Thanks for contributing! :panda_face:

**Comment by ricardogobbosouza** [comment]
Thanks for help!

Agora que vi que você é de São Paulo.
Desculpa meu péssimo inglês!
Obrigado

**Comment by henriquemoody** [comment]
Sem problemas, man! :smile:

**Pull request closed without merging**